client/app/scripts/utils/router-utils.js

```diff
@@ -19,11 +19,18 @@ function encodeURL(url) {
     .replace(new RegExp(SLASH, 'g'), SLASH_REPLACEMENT);
 }
 
-function decodeURL(url) {
+export function decodeURL(url) {
   return decodeURIComponent(url.replace(new RegExp(SLASH_REPLACEMENT, 'g'), SLASH))
     .replace(new RegExp(PERCENT_REPLACEMENT, 'g'), PERCENT);
 }
 
+export function parseHashState(hash = window.location.hash) {
```

Comment:
Nice! Just a tiny observation that `parseStateHash` sounds like a better name to me but it's up to you :)